src/ardrive.ts

```diff
@@ -441,23 +439,29 @@ export class ArDrive extends ArDriveAnonymous {
 		});
 	}
 
-	async getPrivateDrive(driveId: DriveID, drivePassword: string): Promise<ArFSPrivateDrive> {
-		const driveEntity = await this.arFsDao.getPrivateDrive(driveId, drivePassword);
+	async getPrivateDrive(driveId: DriveID, driveKey: DriveKey): Promise<ArFSPrivateDrive> {
+		const driveEntity = await this.arFsDao.getPrivateDrive(driveId, driveKey);
 		return Promise.resolve(driveEntity);
 	}
 
-	async getPrivateFolder(folderId: FolderID, drivePassword: string): Promise<ArFSPrivateFolder> {
-		const folderEntity = await this.arFsDao.getPrivateFolder(folderId, drivePassword);
+	async getPrivateFolder(folderId: FolderID, driveKey: DriveKey): Promise<ArFSPrivateFolder> {
+		const folderEntity = await this.arFsDao.getPrivateFolder(folderId, driveKey);
 		return folderEntity;
 	}
 
-	async getChildrenOfPrivateFolder(folderId: FolderID, password: string): Promise<ArFSPrivateFileOrFolderData[]> {
-		const folder = await this.arFsDao.getPrivateFolder(folderId, password);
+	async getPrivateFolderMetaData(folderId: FolderID): Promise<ArFSPrivateFolderBuilder> {
```

Comment:
A few problems here:
1) This function proclaims to return metadata, but in actuality it returns a builder type. Builders are NOT trustworthy objects meant for data bearing. They are intended to be short-lived objects whose brief lifespan is spent collecting fragmented data in order to build an object. So this is a confusing API that seems easy to use *incorrectly*.
2) The only usage I've found for this in the source is indeed using it incorrectly. It gets a builder (which, at construction time, doesn't do anything except store a few pieces of data and an Arweave reference) and then tries to immediately check for a driveId on it which certainly wouldn't exist yet, and which presumably would throw an exception. And in fact when I tested it, that's exactly what happened:
```
yarn ardrive list-folder  --parent-folder-id 4bf26af6-6b5d-4675-b9a7-11ca634b6d3a --wallet-file ./CLI_TEST_arweave-keyfile-g1hzNXVbh2M6LMQSUYp7HgkgxdadYqYEfw-HAajlms0.json --drive-password "mycoolpassword123"
(node:33978) UnhandledPromiseRejectionWarning: Error: Could not get driveId of private folder
```
The ArDrive layer should probably have nothing to do with ArFS-type builders. Can you please explain what you were trying to accomplish?